lib/Sema/TypeCheckAttr.cpp

```diff
@@ -4553,6 +4564,23 @@ static bool typeCheckDerivativeAttr(ASTContext &Ctx, Decl *D,
 
   // Check access level compatibility for original and derivative functions.
   if (!compatibleAccessLevels()) {
+    // Diagnose if access levels match, but derivative is `@usableFromInline`
+    // while original is not.
+    if (sameAccessLevel && usableFromInlineMismatch) {
+      diags.diagnose(originalName.Loc,
+                     diag::derivative_attr_usable_from_inline_mismatch);
+      // Suggest adding `@usableFromInline` to original.
+      originalAFD
+          ->diagnose(diag::derivative_attr_fix_add_usable_from_inline,
+                     originalAFD->getName())
+          .fixItInsert(
+              originalAFD->getAttributeInsertionLoc(/*forModifier*/ false),
+              "@usableFromInline ");
+      // Suggest remove `@usableFromInline` from derivative.
+      derivative->diagnose(diag::derivative_attr_fix_remove_usable_from_inline,
```

Comment:
This "remove attribute" note isn't entirely accurate: the derivative function may have an `@inlinable` or `@_alwaysEmitIntoClient` attribute instead of `@usableFromInline`.

I think the current message and source location are reasonable though. It seems more fragile to make the diagnostic exactly accurate (checking for one of the three attributes above and putting the location + removal fix-it there).